api/src/timeseries.rs

```diff
@@ -0,0 +1,158 @@
+    Regular(TimeseriesRegular),
+    Irregular(TimeseriesIrregular),
+}
+
+pub async fn get_timeseries_info(
+    conn: &PooledPgConn<'_>,
+    station_id: i32,
+    element_id: String,
+) -> Result<TimeseriesInfo, tokio_postgres::Error> {
+    let ts_result = conn
+        .query_one(
+            "SELECT timeseries.id, \
+                COALESCE(timeseries.fromtime, '1950-01-01 00:00:00+00'), \
+                COALESCE(timeseries.totime, NOW()::timestamptz), \
+                filter.lvl, \
+                filter.sensor, \
+                timeseries.loc \
+                FROM timeseries JOIN labels.filter \
+                    ON timeseries.id = filter.timeseries \
+                WHERE filter.station_id = $1 AND filter.element_id = $2 \
+                LIMIT 1", // TODO: we should probably do something smarter than LIMIT 1
+            &[&station_id, &element_id],
+        )
+        .await?;
+
+    let ts_id: i32 = ts_result.get(0);
+    let fromtime: DateTime<Utc> = ts_result.get(1);
+    // TODO: there might be a better way to deal with totime than that COALESCE
+    let totime: DateTime<Utc> = ts_result.get(2);
+
+    Ok(TimeseriesInfo {
+        ts_id,
+        fromtime,
+        totime,
+        station_id,
+        element_id,
+        lvl: ts_result.get(3),
+        sensor: ts_result.get(4),
+        location: ts_result.get(5),
+    })
+}
+
+pub async fn get_timeseries_data_irregular(
+    conn: &PooledPgConn<'_>,
+    header: TimeseriesInfo,
+    start_time: DateTime<Utc>,
+    end_time: DateTime<Utc>,
+) -> Result<TimeseriesIrregular, tokio_postgres::Error> {
+    let data_results = conn
+        .query(
+            "SELECT obsvalue, obstime FROM data \
+                WHERE timeseries = $1 \
+                    AND obstime BETWEEN $2 AND $3",
+            &[&header.ts_id, &start_time, &end_time],
+        )
+        .await?;
+
+    let ts = {
+        let mut data = Vec::with_capacity(data_results.len());
+        let mut timestamps = Vec::with_capacity(data_results.len());
+
+        for row in data_results {
+            data.push(row.get(0));
+            timestamps.push(row.get(1));
+        }
+
+        TimeseriesIrregular {
+            header,
+            data,
+            timestamps,
+        }
+    };
+
+    Ok(ts)
+}
+
+pub async fn get_timeseries_data_regular(
+    conn: &PooledPgConn<'_>,
+    header: TimeseriesInfo,
+    start_time: DateTime<Utc>,
+    end_time: DateTime<Utc>,
+    time_resolution: String,
+) -> Result<TimeseriesRegular, tokio_postgres::Error> {
+    // TODO: string matching intervals like this is a hack, but currently necessary to avoid
+    // SQL injection. Ideally we could pass an interval type as a query param, which would
+    // also save us the query_string allocation, but no ToSql implementations for intervals
+    // currently exist in tokio_postgres, so we need to implement it ourselves.
+    let interval = match time_resolution.as_str() {
+        "PT1M" => "1 minute",
```

Comment:
PT10M is also pretty standard right now, so maybe that should be here, but its not necessary for the MVP

Reply:
Yeah, in theory we should support arbitrary intervals, which a ToSql implementation will achieve. But I felt it wasn't a priority since: 

- the implementation here would be very Rust-specific, and it's your go version that will actually go into production
- The go code going into production in the short term probably won't support regular timeseries anyway